pkg/kv/kvserver/store_snapshot.go

```diff
@@ -272,17 +272,38 @@ func (kvSS *kvBatchSnapshotStrategy) Receive(
 ) (IncomingSnapshot, error) {
 	assertStrategy(ctx, header, kvserverpb.SnapshotRequest_KV_BATCH)
 
+	// These stopwatches allow us to time the various components of Receive().
+	// - totalTimeStopwatch measures the total time spent within this function.
+	// - batchStopwatch records the time taken to read data out of the gRPC
+	//   stream's batch into memory
+	// - msstwStopwatch measures the time it takes to write the data from the
+	//   snapshot into SSTs
+	// - recvStopwatch records the amount of time spent waiting on the gRPC stream
+	//   and receiving the data from the stream. NB: this value encapsulates wait
+	//   time due to sender-side rate limiting
+	totalTimeStopwatch := timeutil.NewStopWatch()
+	batchStopwatch := timeutil.NewStopWatch()
+	msstwStopwatch := timeutil.NewStopWatch()
+	recvStopwatch := timeutil.NewStopWatch()
+	totalTimeStopwatch.Start()
+
 	// At the moment we'll write at most five SSTs.
 	// TODO(jeffreyxiao): Re-evaluate as the default range size grows.
+	msstwStopwatch.Start()
 	keyRanges := rditer.MakeReplicatedKeyRanges(header.State.Desc)
 	msstw, err := newMultiSSTWriter(ctx, kvSS.st, kvSS.scratch, keyRanges, kvSS.sstChunkSize)
 	if err != nil {
 		return noSnap, err
 	}
 	defer msstw.Close()
+	msstwStopwatch.Stop()
+
+	log.Event(ctx, "Waiting for snapshot batches to begin.")
```

Comment:
[drive by nit] We generally avoid capitalization at the start of these logs + the periods at the end since every particular message is only a fragment of a larger trace (so not quite a sentence).